### PR TITLE
Reworked voting banners

### LINKED ISSUE #1224: 'motion undefined' in voting top bar message
Start motion poll.
Go to front page of a meeting.
Reload.
=> voting message shows "Motion undefined. ..." (motion identifier required)

Workaround: go to motion list -> reload -> motion identifier is visible

### PR DESCRIPTION
**Pull request by luisa-beerboom**
Closes #1224 
Also made the voting banner route to autopilot, if it is available

### CODE REVIEW

**Review by emanuelschuetze — APPROVED** [review]

File: client/src/app/site/pages/meetings/services/active-meeting.service.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ const getMeetingDetailSubscriptionConfig = (id: Id, getNextMeetingIdObservable:
         follow: [
             `chat_group_ids`,
             `chat_message_ids`,
-            { idField: `poll_ids` },
+            { idField: `poll_ids`, follow: [`content_object_id`] },
```
Comment (GabrielInTheWorld) [inline client/src/app/site/pages/meetings/services/active-meeting.service.ts:38]: Why is this necessary?

File: client/src/app/site/pages/meetings/modules/poll/services/vote-controller.service/vote-controller.service.ts [hunk 2]
```diff
@@ -17,4 +17,8 @@ export class VoteControllerService extends BaseMeetingControllerService<ViewVote
     ) {
         super(controllerServiceCollector, Vote, repo);
     }
+
+    public async setHasVotedOnPoll(...viewPolls: ViewPoll[]): Promise<void> {
+        this.repo.setHasVotedOnPoll(...viewPolls);
```
Comment (GabrielInTheWorld) [inline client/src/app/site/pages/meetings/modules/poll/services/vote-controller.service/vote-controller.service.ts:22]: Please put the logic from the repo's function `setHasVotedOnPoll` in this function and make the repo's function `hasVotedFor` public.

**Review by GabrielInTheWorld — CHANGES_REQUESTED** [review]

File: client/src/app/site/pages/meetings/services/active-meeting.service.ts (hunk 1 above)
Comment (luisa-beerboom) [reply client/src/app/site/pages/meetings/services/active-meeting.service.ts:38]: Because I need the contentobject to be loaded to correctly get the data for the banner

File: client/src/app/site/modules/site-wrapper/services/voting-banner.service.ts [hunk 3]
```diff
@@ -86,15 +98,23 @@ export class VotingBannerService {
...
+     * Returns for a given poll a url for the banner.
+     *
+     * @param poll The given poll.
+     *
+     * @returns A string containing the url.
+     */
+    private getUrlForPoll(poll: ViewPoll): string {
+        return this.operator.hasPerms(Permission.meetingCanSeeAutopilot)
+            ? `/${this.activeMeeting.meetingId}/autopilot/`
+            : poll.getContentObjectDetailStateURL();
```
Comment (GabrielInTheWorld) [inline client/src/app/site/modules/site-wrapper/services/voting-banner.service.ts:117]: Please use the function `getDetailStateUrl` and remove the function `getContentObjectDetailStateUrl` from `ViewPoll`s

File: client/src/app/site/modules/site-wrapper/services/voting-banner.service.ts [hunk 4]
```diff
@@ -44,18 +52,13 @@ export class VotingBannerService {
         await this.sendVotesService.setHasVotedOnPoll(...polls);
         // display no banner if in history mode or there are no polls to vote
         const pollsToVote = polls.filter(poll => this.votingService.canVote(poll) && !poll.hasVoted);
```
Comment (GabrielInTheWorld) [inline client/src/app/site/modules/site-wrapper/services/voting-banner.service.ts:54]: Please reference to a class member

File: client/src/app/site/pages/meetings/pages/polls/view-models/has-polls.ts [hunk 5]
```diff
@@ -4,6 +4,7 @@ import { ViewPoll } from './view-poll';
 
 export interface HasPolls<C extends BaseViewModel = any> {
     polls: ViewPoll<C>[];
+    getVotingText: (translateFn: (text: string) => string, poll?: ViewPoll<C>) => string;
```
Comment (GabrielInTheWorld) [inline client/src/app/site/pages/meetings/pages/polls/view-models/has-polls.ts:7]: Why is the poll optional?
Please make an interface describing the context.

File: client/src/app/site/modules/site-wrapper/services/voting-banner.service.ts [hunk 6]
```diff
@@ -86,15 +98,23 @@ export class VotingBannerService {
...
-            return `${motionTranslation} ${contentObject.getNumberOrTitle()}: ${votingOpenedTranslation}`;
-        } else if (poll.isAssignmentPoll) {
-            return `${contentObject.getTitle()}: ${this.translate.instant(`Ballot opened`)}`;
-        } else {
-            return this.translate.instant(`Voting opened`);
-        }
+        return (
+            contentObject?.getVotingText(text => this.translate.instant(text), poll) ??
+            this.translate.instant(`Voting opened`)
+        );
```
Comment (GabrielInTheWorld) [inline client/src/app/site/modules/site-wrapper/services/voting-banner.service.ts:104]:
```suggestion
        if (contentObject) {
            return contentObject.getVotingText(text => this.translate.instant(text), poll);
        } else {
            return this.translate.instant(`Voting opened`);
        }
```

File: client/src/app/site/modules/site-wrapper/services/voting-banner.service.ts [hunk 7]
```diff
@@ -77,6 +80,15 @@ export class VotingBannerService {
         };
     }
 
+    private getBannerCreationData(pollsToVote: ViewPoll[]): BannerCreationData {
+        const isSinglePoll = pollsToVote.length === 1;
```
Comment (GabrielInTheWorld) [inline client/src/app/site/modules/site-wrapper/services/voting-banner.service.ts:84]: Make a class member and access this

File: client/src/app/site/modules/site-wrapper/services/voting-banner.service.ts [hunk 8]
```diff
@@ -77,6 +80,15 @@ export class VotingBannerService {
...
+        const text = isSinglePoll
+            ? this.getTextForPoll(pollsToVote[0])
```
Comment (GabrielInTheWorld) [inline client/src/app/site/modules/site-wrapper/services/voting-banner.service.ts:86]: Make a class member like `firstPoll` or `getFirstPoll()` or `getSinglePoll()`

**Review by GabrielInTheWorld — APPROVED** [review]
Some remarks, but the rest looks good and you can merge it, after resolving them.

**Review by emanuelschuetze — APPROVED** [review]

**Review by GabrielInTheWorld — APPROVED** [review]
